interchain-token-service/src/abi.rs

```diff
@@ -62,10 +61,22 @@ sol! {
     }
 }
 
-impl ItsMessage {
+#[derive(thiserror::Error, Debug, IntoContractError)]
+pub enum Error {
+    #[error("failed to decode ITS message")]
+    MessageDecodeFailed,
+    #[error("invalid message type")]
+    InvalidMessageType,
+    #[error("invalid chain name")]
+    InvalidChainName,
+    #[error("invalid token manager type")]
+    InvalidTokenManagerType,
+}
+
+impl Message {
```

Comment:
I get the difference between `Message` and `HubMessage` but it's still quite confusing that they are put in the same enum in the solidity code.

Reply:
Yeah, it's confusing but we wanted to keep backwards compatibility with existing ITS connections and thus the message format